src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/AvoidDeadConditionalCode_StringContentAnalysis.cs

```diff
@@ -869,5 +869,79 @@ End Sub
             // Test0.vb(6,56): warning CA1508: 'param <> str' is always 'False'. Remove or refactor the condition(s) to avoid dead code.
             GetBasicResultAt(6, 56, "param <> str", "False"));
         }
+
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.StringContentAnalysis)]
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.CopyAnalysis)]
+        [Fact, WorkItem(1650, "https://github.com/dotnet/roslyn-analyzers/issues/1650")]
+        public void StringCompare_InsideConstructorInitializer_Diagnostic()
+        {
+            VerifyCSharp(@"
+class C
+{
+    public bool Flag;
+}
+
+class Base
+{
+    protected Base(bool b) { }
+}
+
+class Test : Base
+{
+    public Test(string s1, string s2, string s3)
+        : base(s1 == s2 && s2 == s3 ? (s1 == s3) : false)
```

Comment:
>s1 == s3 [](start = 39, length = 8)

I assume that this is where the warning actually is? Can you add a `/**/` comment of some kind so it's more easy to visually differentiate? I spent about 20 minutes attempting to figure out how the lower `s1 == s3` was provably true.

Reply:
Yes, and sorry :-) The lower line in the body is just to verify that any data derived from predicates/conditions in the initializer is not carried forward into the method body.

Reply:
I will update the tests with next change in this area.